content/application-service-api.md

```diff
@@ -37,14 +37,8 @@ registration for suspicious regex strings.
 {{% /boxes/note %}}
 
 Application services register "namespaces" of user IDs, room aliases and
-room IDs. These namespaces are represented as regular expressions. An
-application service is said to be "interested" in a given event if one
-of the IDs in the event match the regular expression provided by the
-application service, such as the room having an alias or ID in the
-relevant namespaces. Similarly, the application service is said to be
-interested in a given event if one of the application service's
-namespaced users is the target of the event, or is a joined member of
-the room where the event occurred.
+room IDs. An application service is said to be "interested" in a given event
+if it matches any of the namespaces.
```

Comment:
Not entirely convinced this is an improvement in clarity, personally. The fact that namespaces are defined via regular expressions is a detail which I think is better left for the formal definition below rather in this overview.

(In any case, we'd want to say "namespace regular expressions" rather than "namespace regex patterns").

Reply:
"namespace regular expressions" sounds good to me 👍

I think it makes sense to mention when we're talking about matching. It's immediately apparent and understandable for anyone slightly familiar with regex how it matches then.